src/hues.rs

```diff
@@ -12,81 +14,86 @@ macro_rules! make_hues {
         ///also have some surprising effects if it's expected to act as a
         ///linear number.
         #[derive(Clone, Copy, Debug, Default)]
-        pub struct $name(f32);
+        pub struct $name<T:Float = f32>(T);
 
-        impl $name {
+        impl<T:Float> $name<T> {
             ///Create a new hue from radians, instead of degrees.
-            pub fn from_radians(radians: f32) -> $name {
-                $name(radians * 180.0 / PI)
+            pub fn from_radians(radians: T) -> $name<T> {
+                $name(radians * T::from(180.0).unwrap() / T::from(PI).unwrap())
             }
 
             ///Convert the hue to radians.
-            pub fn to_radians(self) -> f32 {
-                normalize_angle(self.0) * PI / 180.0
+            pub fn to_radians(self) -> T {
+                normalize_angle(self.0) * T::from(PI).unwrap() / T::from(180.0).unwrap()
+            }
+
+            ///Returns the hue value as a f32 or f64
+            pub fn to_float(self) -> T {
+                normalize_angle(self.0)
             }
         }
 
-        impl From<f32> for $name {
-            fn from(degrees: f32) -> $name {
+        impl<T:Float> From<T> for $name<T> {
+            fn from(degrees: T) -> $name<T> {
                 $name(degrees)
             }
         }
 
-        impl Into<f32> for $name {
-            fn into(self) -> f32 {
-                normalize_angle(self.0)
+        impl<T: Float> Into<f64> for $name<T> {
+            fn into(self) -> f64 {
+                normalize_angle(self.0).to_f64().unwrap()
             }
         }
 
-        impl PartialEq for $name {
-            fn eq(&self, other: &$name) -> bool {
-                let hue_s: f32 = (*self).into();
-                let hue_o: f32 = (*other).into();
-                hue_s.eq(&hue_o)
+        impl<T: Float> Into<f32> for $name<T> {
+            fn into(self) -> f32 {
+                normalize_angle(self.0).to_f32().unwrap()
```

Comment:
Can we just restrict these two to be `impl Into<f32> for $name<f32>` and`impl Into<f64> for $name<f64>`, to begin with?